extensions/binding/org.eclipse.smarthome.binding.homematic/README.md

```diff
@@ -0,0 +1,498 @@
+- [piVCCU](https://github.com/alexreinert/piVCCU)
+- [YAHM](https://github.com/leonsio/YAHM)
+
+These ports are used by the binding by default to communicate **TO** the gateway:
+
+- RF components: 2001
+- WIRED components: 2000
+- HMIP components: 2010
+- CUxD: 8701
+- TclRegaScript: 8181
+- Groups: 9292
+
+And **FROM** the gateway to the binding:
+
+- XML-RPC: 9125
+- BIN-RPC: 9126
+
+CCU Autodiscovery:
+* UDP 43439
+
+**Note:** The binding tries to identify the gateway with XML-RPC and uses henceforth:
+
+-   **CCU**
+    - **RF**: BIN-RPC
+    - **WIRED**: BIN-RPC
+    - **HMIP**: XML-RPC
+    - **CUxD**: BIN-RPC (CUxD version >= 1.6 required)
+    - **Groups**: XML-RPC
+
+-   **Homegear**
+    - BIN-RPC
+
+-   **Other**
+    - XML-RPC
+
+## Supported Things
+
+All devices connected to a Homematic gateway.
+All required ESH metadata are generated during device discovery.
+With Homegear or a CCU, variables and scripts are supported too.
+
+## Discovery
+
+Gateway discovery is available:
+* CCU
+* RaspberryMatic >= 2.29.23.20171022
+* Homegear >= 0.6.x
+* piVCCU
+
+For all other gateways you have to manually add a bridge in a things file. Device discovery is supported for all gateways. 
+ 
+The binding has a gateway type autodetection, but sometimes a gateway does not clearly notify the type.
+If you are using a YAHM for example, you have to manually set the gateway type in the bride configuration to CCU.  
+
+If autodetection can not identify the gateway, the binding uses the default gateway implementation.
+The difference is, that variables, scripts and device names are not supported, everything else is the same.
+
+### Automatic install mode during discovery
+
+Besides discovering devices that are already known by the gateway, it may be desired to connect new devices to your system - which requires your gateway to be in install mode. Starting the binding's DiscoveryService will automatically put your gateway(s) in install mode for a specified period of time (see installModeDuration).
+
+**Note:** Enabling / disabling of install mode is also available via GATEWAY_EXTRAS. You may use this if you prefer.
+
+**Exception:** If a gateway is not ONLINE, the install mode will not be set automatically. _For instance during initialization of the binding its DiscoveryService is started and will discover devices that are already connected. However, the install mode is not automatically enabled in this situation because the gateway is in the status INITIALIZING._
+
+## Bridge Configuration
+
+There are several settings for a bridge:
+
+-   **gatewayAddress** (required)  
+Network address of the Homematic gateway
+
+-   **gatewayType**  
+Hint for the binding to identify the gateway type (auto|ccu|noccu) (default = auto).
+
+-   **callbackHost**  
+Callback network address of the ESH runtime, default is auto-discovery
```

Comment:
Avoid "ESH" as it is no official abbreviation and it should not even turn up in solutions documentation.
Best simply remove ESH here.